### DISCUSSION TITLE
Improving performance

### DISCUSSION THREAD

**Discussion by devbar** [OpenVoiceOS/ovos-buildroot#282] [General]
Hi guys, 

first let me say: I really love this project. I am developer and convinced, tech should work for the people, not for the billionairs!

To my question: I made the first steps and made good experiences with a Raspi 3B+ MicHat 2, but the response time will not pass the family-acceptance-test.

I have in mind to distribute the system by using the Raspi as an satelite and moving the backend-end stuff to my home-server-rack, 

- Does anyone have xp with that? 
- Is it a waste of time, because better hardware will not improve to a response time (let me say ~4 seconds or better)?
- What component should I focus on to increase performance: CPU, GPU, Memory, IO ... let's kill my christmas bonus :D

best regards

Benjamin

**Comment by goldyfruit** [comment]
Please have a look to these videos/shorts https://www.youtube.com/@smartgic

I did different tests with different Raspberry Pi. For the Raspberry Pi 3B+ I would recommend to use HiveMind (distributed) to offload the most as you can to a more powerful server.

**Comment by JarbasAl** [comment]
using a local TTS (self hosted or on device) and STT (self hosted) will be much faster than using the default public servers, thats where most of the latency is

if you are willing to sacrifice privacy you can use the chromium STT plugin
